services/cd-service/pkg/repository/transformer.go

```diff
@@ -641,6 +689,28 @@ func (u *UndeployApplication) Transform(ctx context.Context, state *State) (stri
 
 	}
 	// remove application
+	releasesDir := fs.Join(appDir, "releases")
+	files, err := fs.ReadDir(releasesDir)
+	if err != nil {
+		return "", nil, fmt.Errorf("could not read the releases directory: %w", err)
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			releaseDir := fs.Join(releasesDir, file.Name())
+			commitIDFile := fs.Join(releaseDir, "source_commit_id")
+			var commitID string
+			if dat, err := util.ReadFile(fs, commitIDFile); err != nil {
+				// release does not have a corresponding commit, which might be the case if it's an undeploy release, no prob
+			} else {
+				commitID = string(dat)
+				if valid.SHA1CommitID(commitID) {
+					if err := removeCommit(fs, commitID, u.Application); err != nil {
+						return "", nil, fmt.Errorf("could not remove the commit: %w", err)
```

Comment:
i'm not sure if this error contains both the commitId and the app name.
If it does not, please extend it (similar to my suggestion above about "releasesDir")

Reply:
Now the error message from `removeCommit` is extended. This error message is unmodified not to duplicate information, is this okay?

Reply:
Previously the propagated error message did not have neither the commit ID nor the application name, now it has both. This line will remain the same not to duplicate information. Is this acceptable?